src/main/scala/scorex/core/validation/ModifierValidator.scala

```diff
@@ -92,173 +61,122 @@ object ModifierValidator {
     */
-  def validateEqualIds(given: => ModifierId, expected: => ModifierId)
-                      (error: String => Invalid): ValidationState[T] = {
-    validate(given == expected) {
-      error(s"Given: ${e.encodeId(given)}, expected ${e.encodeId(expected)}")
+  def validateEqualIds(id: Short, given: => ModifierId, expected: => ModifierId): ValidationState[T] = {
+    pass {
+      if (!settings.isActive(id) || given == expected) result
+      else settings.getError(id, s"Given: ${e.encodeId(given)}, expected ${e.encodeId(expected)}")
     }
   }
 
   /** Wrap semantic validity to the validation state: if semantic validity was not Valid, then return the `error` given
     */
-  def validateSemantics(validity: => ModifierSemanticValidity)(error: => Invalid): ValidationState[T] = {
-    validateNot(validity == ModifierSemanticValidity.Invalid)(error)
-  }
-
-  /** Validate the condition is `true` or else return the `error` given
-    */
-  def validate(condition: => Boolean)(error: => Invalid): ValidationState[T] = {
-    pass(if (condition) result else error)
-  }
-
-  /** Replace payload with the new one, discarding current payload value. This method catches throwables
-    */
-  def payload[R](payload: => R): ValidationState[R] = {
-    pass(result(payload))
-  }
-
-  /** Map payload if validation is successful
-    */
-  def payloadMap[R](f: T => R): ValidationState[R] = {
-    copy(result = result.map(f))
+  def validateSemantics(id: Short, validity: => ModifierSemanticValidity): ValidationState[T] = {
+    validateNot(id, validity == ModifierSemanticValidity.Invalid)
   }
 
   /** Validate the `condition` is `Success`. Otherwise the `error` callback will be provided with detail
     * on a failure exception
     */
-  def validateNoFailure(condition: => Try[_])(error: Throwable => Invalid): ValidationState[T] = {
-    pass(condition.fold(error, _ => result))
+  def validateNoFailure(id: Short, condition: => Try[_]): ValidationState[T] = {
+    pass(if (!settings.isActive(id)) result else condition.fold(e => settings.getError(id, e), _ => result))
   }
 
   /** Validate the `block` doesn't throw an Exception. Otherwise the `error` callback will be provided with detail
     * on the exception
     */
-  def validateNoThrow(block: => Any)(error: Throwable => Invalid): ValidationState[T] = {
-    validateNoFailure(Try(block))(error)
+  def validateNoThrow(id: Short, block: => Any): ValidationState[T] = {
+    validateNoFailure(id, Try(block))
   }
 
   /** Validate `condition` against payload is `true` or else return the `error`
     */
-  def validateTry[A](tryValue: => Try[A], error: Throwable => Invalid)
-                    (operation: (ValidationState[T], A) => ValidationResult[T]): ValidationState[T] = {
-    pass(tryValue.fold(error, v => operation(this, v)))
+  def validateTry(id: Short, operation: T => Try[T], condition: T => Boolean): ValidationState[T] = {
```

Comment:
`condition: T => Boolean` instead of `operation: (ValidationState[T], A) => ValidationResult[T]` forces checks to be "flat"